cpp/include/cuspatial/experimental/iterator_factory.cuh

```diff
@@ -57,6 +59,18 @@ struct vec_2d_to_tuple {
   }
 };
 
+/**
+ * @internal
+ * @brief Helper to convert a `bouinding_box` into a tuple of elements
+ */
+template <typename T, typename Box = box<T>>
+struct box_to_tuple {
+  __device__ thrust::tuple<T, T, T, T> operator()(Box const& box)
```

Comment:
May be more convenient to just add a conversion operator to `box` to convert to `tuple<T,T,T,T>`?

Reply:
How do I pass that conversion operator to this code instead of `box_to_tuple`? Don't I still need to create a functor for my ``transform_output_iterator`? In that case it wouldn't be any more convenient.

```
template <typename MinXIter, typename MinYIter, typename MaxXIter, typename MaxYIter>
auto make_box_output_iterator(MinXIter min_x, MinYIter min_y, MaxXIter max_x, MaxYIter max_y)
{
  using T         = typename std::iterator_traits<MinXIter>::value_type;
  auto zipped_out = thrust::make_zip_iterator(min_x, min_y, max_x, max_y);
  return thrust::transform_output_iterator(zipped_out, detail::box_to_tuple<T>());
}
```